modules/indigo/module/inc/indigo/of_state_manager.h

```diff
@@ -329,6 +372,44 @@ indigo_core_gentable_register(
 void
 indigo_core_gentable_unregister(indigo_core_gentable_t *gentable);
 
+/*
+ * @brief resume indigo add()/modify() when the driver gets the final status 
+ * @param cxn_id connection id
+ * @param obj of message of this operation
+ * @param priv driver's private data for this entry
+ * @param err_txt error string from driver
+ * @param rv the final status of the driver async add/modify
+ *
+ * The rv status will decide how the indigo to do with the rest works.
+ * For non-async operation, the indigo will call this function internally.
+ */
+
+void
+indigo_core_gentable_entry_add_resume(
+    indigo_cxn_id_t cnx_id,
+    of_object_t *obj,
+    void *priv,
+    of_desc_str_t *err_txt,
```

Comment:
i think this can be "of_desc_str_t err_txt", instead of "of_desc_str_t *err_txt", since "of_desc_str_t XXX" is "char XXX[256]"